djtriggers/tests/test_models.py

```diff
@@ -67,6 +69,64 @@ def test_handle_execution_failure_tries_exceeded(self, mock_logger):
                                                                              try_count=original_tries+1)
         mock_logger.assert_called_with(trigger, message, level=WARNING)
 
+    @override_settings(DJTRIGGERS_TRIES_BEFORE_WARNING=3)
+    @override_settings(DJTRIGGERS_TRIES_BEFORE_ERROR=5)
+    def test_handle_execution_failure_tries_limit(self):
+        """
+        Check behavior when trigger reaches retries limit and ERROR is raised
+        """
+        exception = Exception()
+        exception_name = type(exception).__name__
+        trigger = DummyTriggerFactory(number_of_tries=3)
+        original_tries = trigger.number_of_tries
+
+        with patch.object(TriggerLogger, 'log_message') as mock_logger:
+            trigger._handle_execution_failure(exception)
+            exceeded_retries = original_tries + 1
+            assert trigger.number_of_tries == exceeded_retries
+            assert trigger.successful is None
+            message = ('Processing of {trigger_type} {trigger_key} '
+                       'raised a {exception_type} (try nr. {try_count})').format(trigger_type=trigger.trigger_type,
+                                                                                 trigger_key=trigger.pk,
+                                                                                 exception_type=exception_name,
+                                                                                 try_count=exceeded_retries)
+            mock_logger.assert_called_with(trigger, message, level=WARNING)
+
+        # do an extra retry
+        with patch.object(TriggerLogger, 'log_message') as mock_logger:
+            trigger._handle_execution_failure(exception)
+            exceeded_retries = original_tries + 2
+            assert trigger.number_of_tries == exceeded_retries
+            assert trigger.successful is False
+            message = ('Processing of {trigger_type} {trigger_key} '
+                       'raised a {exception_type} (try nr. {try_count})').format(trigger_type=trigger.trigger_type,
+                                                                                 trigger_key=trigger.pk,
+                                                                                 exception_type=exception_name,
+                                                                                 try_count=exceeded_retries)
+            mock_logger.assert_called_with(trigger, message, level=ERROR)
```

Comment:
Would it make sense to use `assert_called_once_with` instead? I assume the expected behavior is that exactly one logging call on that level happens.

Reply:
done